### ISSUE TITLE
feature: pass custom tslint ignores

### ISSUE THREAD

**Issue by zewa666** [metadevpro/ts-pegjs#1]
finally a plugin to create typescript results out of pegs 🎉 

It'd be great if we could pass custom tslintIgnoreRules via the CLI to suite whatever settings the consumer has.

So here https://github.com/metadevpro/ts-pegjs/blob/master/src/passes/generate-ts.js#L1233
we could instead do something like:

```typescript
function generateGeneratedByComment() {
      return [
        "// Generated by PEG.js v. " + pegJsVersion + " (ts-pegjs plugin v. " + pluginVersion + " )",
        "//",
        "// https://pegjs.org/   https://github.com/metadevpro/ts-pegjs",
        "",
        ...options.tslintIgnores
          ? options.tslintIgnores.map((rule) => "// tslint:disable:" + rule)
          : [
            "// tslint:disable:only-arrow-functions",
            "// tslint:disable:object-literal-shorthand",
            "// tslint:disable:trailing-comma",
            "// tslint:disable:object-literal-sort-keys",
            "// tslint:disable:one-variable-per-declaration",
            "// tslint:disable:max-line-length",
            "// tslint:disable:no-consecutive-blank-lines",
            "// tslint:disable:align"
          ],
        options.trace ? "// tslint:disable:no-console" : ""
      ].join("\n");
    }
```

to either keep your locals or provide an array of rules (eg string separated) via the CLI.

I wasn't sure though how to pass new cli attributes, as even the mentioned example with `--cache` results in a `Too many arguments.` error

**Comment by pjmolina** [comment]
Thanks for point the problem out.
The main problem I see is this solution (passing a full list of rules to avoid) it is do not scales well with the number of rules.
I am thinking in providing the following alternative:

1. Improve the default linting exclusions also for including the `--cache` use case.
2. A cli parameter `--no-tslint` to avoid all these default exclusions. And then, you can optionally put any as needed as a pegjs initializer (inside your grammar in the bracket preamble).

Will this proposed solution work for your use case?

**Comment by zewa666** [comment]
well the thing is that we need to make sure that the tslint ignores are at the top (comment section) of the file in order to catch linting issues with default stuff generated by peg like peg$parse ... As far as I got the pegjs initializer section will be placed later on right?

But yeah having a `--no-tslint` option would definitely help in being able to setup my own linting excludes.

**Comment by pjmolina** [comment]
You are right: initializers would not help here if you need in the top header.
Would it help your use case if move the default tslint exclusions to the head of the file?

Anyway, I would start adding the `--no-tslinit` asap.

**Comment by pjmolina** [comment]
Hi again. Just implemented:
- `no-tslint` + `tslint-ignores` options.
- moved the tslint exclusions to the top of the file.

Changes are in the [tslint-opt](https://github.com/metadevpro/ts-pegjs/tree/tslint-opt) branch.
If you don't found any bug or stopper I can merge it with master and release a new version with it.

**Comment by zewa666** [comment]
@pjmolina thanks that works like a charm. I've just created a PR to add some more information on how to properly use `--extra-options-file`, which can reduce some pain due to misplaced argument location :)

**Comment by pjmolina** [comment]
Excelent! 
Thank you for your help @zewa666 
Merged and published as npm package version [0.1.8](https://www.npmjs.com/package/ts-pegjs).

**Comment by zewa666** [comment]
thanks for the lightning fast response and update 👍 
updated, configured and everything works ... no more screaming linter yeey
